src/runtime/virtcontainers/kata_agent.go

```diff
@@ -1226,13 +1261,69 @@ func (k *kataAgent) rollbackFailingContainerCreation(ctx context.Context, c *Con
 			k.Logger().WithError(err2).Error("rollback failed unmountHostMounts()")
 		}
 
-		if err2 := bindUnmountContainerRootfs(ctx, getMountPath(c.sandbox.id), c.id); err2 != nil {
-			k.Logger().WithError(err2).Error("rollback failed bindUnmountContainerRootfs()")
+		if c.rootFs.Type == NydusRootFSType {
+			if err2 := nydusContainerCleanup(ctx, getMountPath(c.sandbox.id), c); err2 != nil {
+				k.Logger().WithError(err2).Error("rollback failed nydusContainerCleanup")
+			}
+		} else {
+			if err2 := bindUnmountContainerRootfs(ctx, getMountPath(c.sandbox.id), c.id); err2 != nil {
+				k.Logger().WithError(err2).Error("rollback failed bindUnmountContainerRootfs()")
+			}
 		}
 	}
 }
 
+func (k *kataAgent) buildContainerRootfsWithNydus(sandbox *Sandbox, c *Container, rootPathParent string) (*grpc.Storage, error) {
+	if sandbox.GetHypervisorType() != string(QemuHypervisor) {
+		return nil, errors.New("nydus only support qemu currently")
```

Comment:
This is the same error as the one in `mount.go`. It would be good to avoid the duplication, and ideally find a way to only return that (single) error from within `nydus.go` somehow.

Reply:
done, add err type `errNydusdNotSupport`